platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -49,7 +64,81 @@ public static String formatNodeName(final long nodeId) {
      * @return its Hash
      */
     @NonNull
-    public static Hash hash(@NonNull final Roster roster) {
+    public static Hash hashOf(@NonNull final Roster roster) {
         return PBJ_RECORD_HASHER.hash(roster, Roster.PROTOBUF);
     }
+
+    /**
+     * Creates a new roster from the given address book.
+     *
+     * @param addressBook the address book
+     * @return a new roster
+     */
+    @NonNull
+    public static Roster createRoster(@NonNull final AddressBook addressBook) {
```

Comment:
How does this method differ from `RosterRetriever::buildRoster`?

Reply:
Just looked, it looks like they're the same though this version (moved from ABInitializer to closer reflect a roster util that it is) appears to have come first (and also used by all Mains). In order not to further complicate the problem though, I'm reverting RosterUtils changes. One of the two methods can be removed in a later PR.

Reply:
filed an issue https://github.com/hashgraph/hedera-services/issues/15573

Reply:
.